### DISCUSSION TITLE
How to pass sqlx pool into services implementation?

### DISCUSSION THREAD

**Discussion by mpiorowski** [hyperium/tonic#1264] [Q&A] [answered]
How to pass postgress sqlx connection pool into services implementation? Is there some kind of Context like in Go? 

Would be rly greatfull for help.

**Comment by LucioFranco** [comment, marked as answer]
You would place it in this struct https://github.com/hyperium/tonic/blob/master/examples/src/helloworld/server.rs#L11 and it will be available to each handler via the `&self`. If you need to mutate it then you will need to figure out how to do that via a `Mutex` etc.

**Comment by mpiorowski** [comment]
Thank You :)
